immer/detail/iterator_facade.hpp

```diff
@@ -52,12 +52,12 @@ template <typename DerivedT,
           typename DifferenceTypeT = std::ptrdiff_t,
           typename PointerT = T*>
 class iterator_facade
-    : public std::iterator<IteratorCategoryT,
-                           T,
-                           DifferenceTypeT,
-                           PointerT,
-                           ReferenceT>
```

Comment:
What is the purpose of this change?

Reply:
**High order bit:** I'd defer to you on good/bad on that. Don't have any opinions/was only cleaning code/l

Motivation was purely getting rid of MSVC deprecation warnings (MSVC 2019, 16.5.1).
In the middle of a different branch/repo, but IIRC, this was the specific warning:

> D.13 [depr.iterator.primitives]: “warning STL4015: The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.”

[This https://devblogs.microsoft.com/cppblog/c17-feature-removals-and-deprecations/](This https://devblogs.microsoft.com/cppblog/c17-feature-removals-and-deprecations/)

Other option is to #define  _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS  for MSVC targets

Reply:
I agree with your change then, thank you!